contracts/exchangeIssuance/IssueRedeem.sol

```diff
@@ -0,0 +1,192 @@
+pragma solidity >=0.6.10;
+pragma experimental ABIEncoderV2;
+
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import "@uniswap/v2-periphery/contracts/interfaces/IUniswapV2Router02.sol";
+import "@uniswap/v2-core/contracts/interfaces/IUniswapV2Factory.sol";
+import "../lib/UniswapV2Library.sol";
+import "../lib/SushiswapV2Library.sol";
+import "../interfaces/ISetToken.sol";
+import "../interfaces/IBasicIssuanceModule.sol";
+import "../interfaces/IWETH.sol";
+
+/**
+ * @title IssueRedeem
+ * @author Noah Citron
+ *
+ * Contract that allows for the minting and redeeming any Set token using
+ * ETH as the paying/receiving currency. All swaps are done using the best price
```

Comment:
ETH is a good start, though I imagine that people will want to pay/receive other currencies like USDC, DAI, etc. - which should be part of the spec.